pkg/reconciler/ingress/ingress.go

```diff
@@ -43,33 +44,44 @@ const (
 	preferSyncValue         = "respond-sync"
 	asyncAlwaysMode         = "always.async.knative.dev"
 	asyncConditionalMode    = "conditional.async.knative.dev"
-	publicLBDomain          = "istio-ingressgateway.istio-system.svc.cluster.local"
-	privateLBDomain         = "knative-local-gateway.istio-system.svc.cluster.local"
+	publicLBDomain          = "knative-local-gateway.istio-system.svc.cluster.local"
+	privateLBDomain         = "istio-ingressgateway.istio-system.svc.cluster.local"
 	producerServiceName     = "async-producer"
 	asyncOriginalHostHeader = "Async-Original-Host"
 	ingressClassName        = "INGRESS_CLASS_NAME"
 	ingressSuffix           = ".ingress.networking.knative.dev"
 )
 
+type loadBalancerDomain struct {
+	Private, Public string
+}
+
+var loadBalancers = map[string]loadBalancerDomain{
+	"istio":   loadBalancerDomain{"istio-ingressgateway.istio-system.svc.cluster.local", "knative-local-gateway.istio-system.svc.cluster.local"},
+	"kourier": loadBalancerDomain{"kourier.kourier-system.svc.cluster.local", "kourier.kourier-system.svc.cluster.local"},
+	// "contour":    loadBalancerDomain{"",""},
+	// "ambassador": loadBalancerDomain{"",""}, TODO Add contour/ambassador after successful tests in cluster
+}
+
 // ReconcileKind implements Interface.ReconcileKind.
 func (r *Reconciler) ReconcileKind(ctx context.Context, ing *v1alpha1.Ingress) reconciler.Event {
 	logger := logging.FromContext(ctx)
 	ingressClass := os.Getenv(ingressClassName)
 
-	if strings.HasSuffix(ingressClass,ingressSuffix) {
-		logger.Debugf("valid ingress suffix detected, using ingress class name %s", ingressClass)
+	if ingressName, ok := loadBalancers[strings.Split(ingressClass, ".")[0]]; ok {
+		logger.Debugf("valid ingress name detected %s, using ingress class %s", ingressName, ingressClass)
```

Comment:
I would not leave debug logs in the code. Find for info and your debugging but not to leave there in official PR